src/client/thirdParty/Amplitude.tsx

```diff
@@ -36,33 +37,51 @@ type AmplitudeProps = {
   apiKey: string;
 };
 
-export const Amplitude: FunctionComponent<AmplitudeProps> = ({ apiKey }) => {
+function initAmplitude(apiKey: string) {
+  amplitude.init(apiKey, {
+    defaultTracking: {
+      attribution: false,
+      sessions: false,
+      formInteractions: false,
+      fileDownloads: false,
+    },
+    serverUrl: AMPLITUDE_INGRESS_PROXY,
+  });
+}
+
+function usePlaygroundSignalsWrapper() {
   usePlaygroundSignals({
     onServerApiSuccess: (event) => {
       const visitorId = event.products.identification?.data?.visitorId;
       const botDetected = event?.products?.botd?.data?.bot?.result === 'bad' ? 'True' : 'False';
 
       amplitude.add(demoPageViewedEventPropertiesEnrichment(botDetected));
-      amplitude.init(apiKey, {
-        defaultTracking: {
-          pageViews: {
-            eventType: EVENT_TYPE,
-          },
-          attribution: false,
-          sessions: false,
-          formInteractions: false,
-          fileDownloads: false,
-        },
-        deviceId: visitorId,
-        serverUrl: AMPLITUDE_INGRESS_PROXY,
-      });
 
-      // Set Amplify user's custom `botDetected` property based on Fingerprint Bot Detection result
-      const identifyEvent = new amplitude.Identify();
-      identifyEvent.set('botDetected', botDetected);
-      amplitude.identify(identifyEvent);
+      amplitude.track(EVENT_TYPE, {
+        botDetected,
+        visitorId,
+      });
     },
   });
+}
+
+export async function trackAskAIkHelpMethodChosen(
+  helpMethod: string,
+  visitorId: string,
+  pagePath: string,
+  pageTitle: string,
+) {
+  amplitude.track(ASK_AI_CHOSEN_EVENT_TYPE, {
+    helpMethod,
+    visitorId,
+    'Docs Page Path': pagePath,
+    'Docs Page Title': pageTitle,
+  });
+}
+
+export const Amplitude: FunctionComponent<AmplitudeProps> = ({ apiKey }) => {
+  initAmplitude(apiKey);
```

Comment:
You cannot initialize Amplitude in a component render method

1. It runs at build time on the server, causing errors.
2. At runtime, It runs needlessly many times on every component render.


<img width="976" alt="image" src="https://github.com/user-attachments/assets/276abf38-b3ef-4f39-a0ed-6f0f6d4506a1">

<img width="292" alt="image" src="https://github.com/user-attachments/assets/ddb58d23-76db-49fb-90d3-b0c0d64118db">

Reply:
Reverted amplitude changes